src/Host/Monitor/Impl/BrokerManager.cs

```diff
@@ -10,16 +11,16 @@ namespace Microsoft.R.Host.Monitor {
     public class BrokerManager {
```

Comment:
Testing this will be very painful, because it's tightly coupled to MainWindow UI. I would recommend not accessing `MainWindow` directly anywhere, and instead providing a generic callback for status updates and such. The caller of `CreateOrAttachToBrokerInstanceAsync` would then be responsible for translating those to status bar updates etc; and in tests, we'd use a mock to check that progress is reported correctly.